pkg/archive/archive_test.go

```diff
@@ -1318,3 +1320,74 @@ func readFileFromArchive(t *testing.T, archive io.ReadCloser, name string, expec
 	assert.NoError(t, err)
 	return string(content)
 }
+
+type fakeHook struct {
+	entries  []*logrus.Entry
+	disabled bool
+}
+
+func newFakeHook() *fakeHook {
+	return &fakeHook{entries: make([]*logrus.Entry, 0)}
+}
+
+func (fh *fakeHook) Levels() []logrus.Level {
+	if !fh.disabled {
+		return []logrus.Level{logrus.DebugLevel}
+	}
+	return []logrus.Level{}
+}
+
+func (fh *fakeHook) Fire(entry *logrus.Entry) error {
+	if !fh.disabled {
+		fh.entries = append(fh.entries, entry)
+	}
+	return nil
+}
+
+func (fh *fakeHook) disable() {
+	fh.disabled = true
+}
+
+func TestDisablePigz(t *testing.T) {
+	os.Setenv("DISABLE_PIGZ", "true")
+	defer os.Unsetenv("DISABLE_PIGZ")
+
+	logs := getDecompressLogs(t)
+	require.Len(t, logs, 0)
+}
+
+func TestPigz(t *testing.T) {
+	_, err := exec.LookPath("unpigz")
+	if err != nil {
+		t.Skip("Cannot run test, unpigz not installed")
+	}
+
+	logs := getDecompressLogs(t)
+	require.Len(t, logs, 1)
+	assert.Contains(t, logs[0].Message, "Decompressing with unpigz")
```

Comment:
Please don't use log messages to test this.

Reply:
What else would you suggest doing?

Reply:
Not using log messages to test the functionality. I am not sure what these are testing exactly, so I don't have a good suggesting. Testing with log message hooks isn't great, as the log messages may change.

Reply:
I'm trying to test if pigz was used, and if it can be disabled. The alternative is to pass in some callback or something that's only used for testing which returns whether pigz was used or !pigz.